plugin/evm/vm.go

```diff
@@ -625,6 +600,42 @@ func (vm *VM) Initialize(
 	return vm.initializeStateSyncClient(lastAcceptedHeight)
 }
 
+func parseGenesis(ctx *snow.Context, bytes []byte) (*core.Genesis, error) {
+	g := new(core.Genesis)
+	if err := json.Unmarshal(bytes, g); err != nil {
+		return nil, fmt.Errorf("parsing genesis: %w", err)
+	}
+
+	// Populate the Avalanche config extras.
+	configExtra := params.GetExtra(g.Config)
+	configExtra.AvalancheContext = extras.AvalancheContext{
+		SnowCtx: ctx,
+	}
+
+	// TODO: This check should be removed. The upgrade config is always provided
+	// in production. All tests should be updated to correctly provide the
+	// upgrade config.
+	if ctx.NetworkUpgrades != (upgrade.Config{}) {
+		configExtra.NetworkUpgrades = extras.GetNetworkUpgrades(ctx.NetworkUpgrades)
+	}
+
+	// If Durango is scheduled, schedule the Warp Precompile at the same time.
+	if configExtra.DurangoBlockTimestamp != nil {
```

Comment:
Should this handle the unscheduled time?

Reply:
I don't think it matters. This PR is only really needed because of some enforced ordering in the eth config.